packages/lisk-core/src/controller.js

```diff
@@ -1,40 +1,171 @@
+			this.config.dirs.root
+		})`;
 
-		await Promise.map(Object.keys(this.modules), async (m) => {
-			await this.modules[m].load();
-		});
+		// Make sure all directories exists
+		fs.emptyDirSync(this.config.dirs.temp);
+		Object.keys(this.config.dirs).forEach(dir =>
+			fs.ensureDirSync(this.config.dirs[dir]),
+		);
+		fs.writeFileSync(`${this.config.dirs.pids}/controller.pid`, process.pid);
+
+		// Set the custom directory to load modules
+		if (this.config.dirs.modules) {
+			logger.info(
+				`Setting custom module directory to: ${this.config.dirs.modules}`,
+			);
+		}
 
-		console.info(`Bus listening to events: ${this.bus.getEvents()}`);
-		console.info(`Bus ready for actions: ${this.bus.getActions()}`);
+		// Setup bus RPC socket
+		await this.bus.setup();
+	}
+
+	async load() {
+		await this.establishChannel();
+		await this.loadModules();
+
+		logger.info('Bus listening to events', this.bus.getEvents());
+		logger.info('Bus ready for actions', this.bus.getActions());
+
+		this.channel.publish('lisk:ready', {});
 	}
 
 	getModule(alias) {
 		return this.modules[alias];
 	}
+
+	async establishChannel() {
+		this.channel = new EventEmitterChannel(
+			'lisk',
+			['ready'],
+			['getComponentConfig'],
+			this.bus,
+			{ skipInternalEvents: true },
+		);
+
+		await this.channel.registerToBus();
+
+		this.channel.action(
+			'getComponentConfig',
+			async action => this.config.components[action.params],
+		);
+
+		// If log level is greater than info
+		if (logger.level() < 30) {
+			this.bus.onAny((name, event) => {
+				logger.debug(
+					`MONITOR: ${event.source} -> ${event.module}:${event.name}`,
+					event.data,
+				);
+			});
+		}
+	}
+
+	async loadModules() {
+		this.config.modules.forEach(moduleConfig => {
+			const npmPackage = this.config.dirs.modules
+				? `${this.config.dirs.modules}/${moduleConfig.npmPackage}`
+				: moduleConfig.npmPackage;
+			const module = ModuleFactory.create(
+				moduleConfig.loadAs,
+				npmPackage,
+				moduleConfig.options,
+				logger,
+				this.bus,
+			);
+			this.modules[module.alias] = module;
+		});
+
+		return Promise.each(Object.keys(this.modules), m => this.modules[m].load());
+	}
+
+	async unloadModules(modules = null) {
+		return Promise.mapSeries(modules || Object.keys(this.modules), async m => {
+			await this.modules[m].unload();
+			delete this.modules[m];
+		});
+	}
+
+	static async start(options = {}) {
+		logger.info('Starting lisk...');
+		const ctrl = new Controller(Object.assign({}, config, options));
+		await ctrl.setup();
+
+		process.on('uncaughtException', err => {
+			// Handle error safely
+			logger.fatal('System error', { message: err.message, stack: err.stack });
+			ctrl.stop(err, 1);
+		});
+
+		process.on('unhandledRejection', err => {
+			// Handle error safely
+			logger.fatal('System error', { message: err.message, stack: err.stack });
+			ctrl.stop(err, 1);
+		});
+
+		process.once('SIGTERM', () => ctrl.stop());
```

Comment:
I think, we should move SIGNAL handlings to where we start the application means cli/index.js. All modules should provide a cleanup method to call when we receive the signal. 

I see that child process channel and child process module also have their own handlers. I prefer to move the  handlers where we initialise the child process to modules/child_process.js maybe?